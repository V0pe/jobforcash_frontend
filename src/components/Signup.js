/* eslint-disable react/jsx-props-no-spreading */
import { React, useState } from 'react';
import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import { signupUser } from '../redux/actions/auth';
import './components.css';

const Signup = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [error, setError] = useState(null);
  const { register, handleSubmit, formState: { errors } } = useForm();

  const onFormSubmit = (data) => {
    dispatch(signupUser(data)).catch(() => setError('Invalid credentials. Try again'));
    navigate('/');
  };

  return (
    <main className="background d-flex flex-row justify-content-center align-items-center">
      <div className="">
        <div className="">
          <h2 className="text-white text-center">SIGN UP FOR AN ACCOUNT</h2>
          <hr className="tiny-line" />
        </div>
        {error && <p className="">{error}</p>}
        {errors.username && <p className="">Username is required</p>}
        {errors.email && <p className="">Email is required</p>}
        {errors.password && <p className="">Password is required</p>}
        <form className="" onSubmit={handleSubmit(onFormSubmit)}>
          <div className="mb-3">
            <input
              className="form-control"
              type="text"
              placeholder="Username"
              {...register('username', { required: 'Username is required' })}
            />
          </div>
          <div className="mb-3">
            <input
              className="form-control"
              type="email"
              placeholder="Email"
              {...register('email', { required: 'Email is required' })}
            />
          </div>
          <div className="mb-3">
            <input
              className="form-control"
              type="password"
              placeholder="Password"
              {...register('password', { required: 'Password is required' })}
            />
          </div>
          <div className="d-flex justify-content-center">
            <input className="btn custom-lime rounded my-3 mx-3" type="submit" value="Sign Up" />
            <Link className="btn btn-success rounded my-3 mx-3" to="/login">Log In</Link>
          </div>
        </form>
      </div>
    </main>
  );
};

export default Signup;
