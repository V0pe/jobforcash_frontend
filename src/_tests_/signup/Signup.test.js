import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Signup from '../../components/Signup';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Signup />
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
