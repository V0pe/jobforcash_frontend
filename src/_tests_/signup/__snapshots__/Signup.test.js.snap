// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="background d-flex flex-row justify-content-center align-items-center"
>
  <div
    className=""
  >
    <div
      className=""
    >
      <h2
        className="text-white text-center"
      >
        SIGN UP FOR AN ACCOUNT
      </h2>
      <hr
        className="tiny-line"
      />
    </div>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <input
          className="form-control"
          name="username"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Username"
          type="text"
        />
      </div>
      <div
        className="mb-3"
      >
        <input
          className="form-control"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
      </div>
      <div
        className="mb-3"
      >
        <input
          className="form-control"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="d-flex justify-content-center"
      >
        <input
          className="btn custom-lime rounded my-3 mx-3"
          type="submit"
          value="Sign Up"
        />
        <a
          className="btn btn-success rounded my-3 mx-3"
          href="/login"
          onClick={[Function]}
        >
          Log In
        </a>
      </div>
    </form>
  </div>
</main>
`;
